source/features/clone-branch.tsx

```diff
@@ -0,0 +1,101 @@
+import delegate, {DelegateEvent} from 'delegate-it';
+import gitBranch from 'octicon/git-branch.svg';
+import insertTextTextarea from 'insert-text-textarea';
+import React from 'dom-chef';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getRepoURL, getRepoGQL} from '../libs/utils';
+
+const getBranchBaseSha = async (branchName: string): Promise<string> => {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			ref(qualifiedName: "${branchName}") {
+				target {
+					oid
+				}
+			}
+		}
+	`);
+
+	return repository.ref.target.oid;
+};
+
+async function cloneBranch(event: DelegateEvent<MouseEvent, HTMLButtonElement>): Promise<void> {
+	const cloneButton = event.delegateTarget;
+	const branchElement = cloneButton.closest<HTMLElement>('[data-branch-name]')!;
+
+	const currentBranch = getBranchBaseSha(branchElement.dataset.branchName!);
+	let newBranchName = prompt('Enter the new branch name')?.trim();
+
+	const loadingIcon = select('.js-loading-spinner', branchElement)!;
+	while (newBranchName) {
+		loadingIcon.hidden = false;
+		cloneButton.hidden = true;
+
+		const result = await createBranch(newBranchName, await currentBranch); // eslint-disable-line no-await-in-loop
+	
+		loadingIcon.hidden = true;
+		cloneButton.hidden = false;
```

Comment:
If you want to fix the loading position:
```suggestion
	const loadingIcon = select('.js-loading-spinner', branchElement)!.cloneNode(true);
	while (newBranchName) {
		cloneButton.replaceWith(loadingIcon);

		// eslint-disable-next-line no-await-in-loop
		const result = await createBranch(newBranchName, await currentBranch); 

		loadingIcon.replaceWith(cloneButton);
```

Thoughts?

Reply:
I would import the icon for 2 reasons.
1. Don't have to worry about github changing the position and creating an error. 
2 easier to read

Reply:
Or you could create a `libs/icon-loading.tsx` file:

```tsx
export default function loadingIcon(): HTMLImageElement {
	return (
		...
	);
}

Reply:
its a gif link

Reply:
@fregante its a gif I should just return the link?

Reply:
or this https://gist.github.com/silverwind/7992679

Reply:
I just went with clone